File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
import sys
sys.setrecursionlimit(100000)

total_length = 0
answer = 0
def DFSRecursive(numbers, target, index, result):
    
    global total_length
    global answer
    
    if index == total_length:
        if result == target:
            answer += 1
        return 

    DFSRecursive(numbers, target, index + 1, result + numbers[index])
    DFSRecursive(numbers, target, index + 1, result - numbers[index])
    

def solution(numbers, target):
    global total_length
    global answer
    total_length = len(numbers)
    
    DFSRecursive(numbers,target,0,0)
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    
    public int answer = 0; 
    
    public void dfsRecursive(int[] numbers,int target, int index, int sum){
        if (index == numbers.length) {
            if (sum == target) {
                answer += 1;
            }
            return;
        }
        
        dfsRecursive(numbers,target,index + 1, sum + numbers[index]);

        dfsRecursive(numbers,target,index + 1, sum - numbers[index]);
    }
    
    public int solution(int[] numbers, int target) {
    
        dfsRecursive(numbers,target,0,0);
        return answer;
    }
}